### ISSUE TITLE
[Phase 2] Feature Engineering and Preprocessing

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/particle-collision-classifier#3]
## Objective

Implement feature preprocessing pipeline and exploratory data analysis for understanding the physics-based features in HEP datasets.

**Part of:** #1

## Dependencies

- #2 (Data Infrastructure) must be complete

### Feature Preprocessing
- [ ] Create `src/features/preprocessor.py` with sklearn Pipeline
- [ ] Implement StandardScaler for feature normalization
- [ ] Handle any missing values with imputation strategy

### Feature Engineering
- [ ] Create `src/features/engineering.py` for derived features
- [ ] Add physics-informed feature combinations (optional)
- [ ] Document feature meanings from dataset documentation

### Exploratory Data Analysis
- [ ] Create `notebooks/01_eda.ipynb`
- [ ] Analyze class distribution (signal vs background)
- [ ] Generate feature correlation heatmap
- [ ] Plot feature distributions by class

### Preprocessor Pipeline

```python
# src/features/preprocessor.py
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.impute import SimpleImputer

class Preprocessor:
    def __init__(self):
        self.pipeline = Pipeline([
            ('imputer', SimpleImputer(strategy='median')),
            ('scaler', StandardScaler())
        ])

    def fit(self, X):
        self.pipeline.fit(X)
        return self

    def transform(self, X):
        return self.pipeline.transform(X)

    def fit_transform(self, X):
        return self.pipeline.fit_transform(X)
```

### EDA Notebook Structure

```python
# notebooks/01_eda.ipynb cells:

# 1. Load Data
from src.data.loader import HEPDataLoader
loader = HEPDataLoader("higgs", sample_size=50000)
X, y = loader.load()

# 2. Class Distribution
import matplotlib.pyplot as plt
y.value_counts().plot(kind='bar')
plt.title('Class Distribution')

# 3. Feature Correlations
import seaborn as sns
corr = X.corr()
plt.figure(figsize=(12, 10))
sns.heatmap(corr, cmap='coolwarm')

# 4. Feature Distributions
fig, axes = plt.subplots(4, 7, figsize=(20, 12))
for i, col in enumerate(X.columns[:28]):
    ax = axes[i//7, i%7]
    X[col].hist(ax=ax, bins=50)
    ax.set_title(col)
```

## Definition of Done

- [ ] Preprocessor produces consistent output shape
- [ ] Pipeline is serializable (can save/load with joblib)
- [ ] EDA notebook runs without errors
- [ ] Feature correlations documented

## Technical Notes

- Features in HIGGS dataset:
  - 21 low-level kinematic features (detector measurements)
  - 7 high-level derived features (physics quantities)
- Signal events are ~47% in HIGGS, more imbalanced in SUSY
